quickwit/quickwit-search/src/root.rs

```diff
@@ -1393,9 +1393,10 @@ async fn assign_client_fetch_docs_jobs(
 }
 
 // Measure the cost associated to searching in a given split metadata.
-fn compute_split_cost(_split_metadata: &SplitMetadata) -> usize {
+fn compute_split_cost(split_metadata: &SplitMetadata) -> usize {
     // TODO: Have a smarter cost, by smoothing the number of docs.
-    1
+    // +1 because a split of 1 doc would have an ilog2 of 0 otherwise
+    split_metadata.num_docs.ilog2() as usize + 1
```

Comment:
2 points...

First point: 
I know `num_docs` is not supposed to be 0 today, but there is really no reason to not be a tiny bit defensive here and make sure for `i = 0`.
The above would be better as :
`split_metadata.num_docs.max(2).ilog2() as usize`

Second point:
I don't the function is good here.
For instance, on both side of the range it behaves wrongly.
In particular, I feel like:
- a split of size 1's cost is way too low compared to a split of size 1024.
- a split of size 10M cost is way too low compared to a split of size 10M.

Of course, as we discussed offline, this is very dependant on the query type and what is in cache or not. It feels uncomfortable to even think about a one size fits all solution.

(it almost make sense to prefer diversity in size for all of the leaf in the algorithm...)

I suggest we do something like 
`K + num_docs / 100_000` ?

With some value of K somewhere between 1 and 5.